lib/ansible/galaxy/role.py

```diff
@@ -400,4 +400,14 @@ def requirements(self):
 
                     break
 
+            if type(self._requirements) is dict:
+                display.warning("Wrong format for requirements for role %s" % self.name)
+                if 'collections' in self._requirements:
+                    display.warning("Roles can not depend on collections. Collections will not be installed")
```

Comment:
I wonder if this should just be an error. Note that this wouldn't really deter a standalone role from relying on an installed collection's role, since the standalone role could just have the FQCN of the collection's role in the requirements.yml.

Reply:
Seems a valid solution too

[Everything is better then an error about dicts and lists]

Reply:
Used an error message now

Reply:
@s-hertel the basic error was technically in the parsing of the `requirements.yml` 

Roles use the "old" format which is a list of dicts while collections use a dictionary with lists of dictionaries.

This just created a rather strange error on a type mismatch without any information for the user

[And I just now understood your remark on the usage of the FQCN fully ]